### PR TITLE
[cmake] update OT_LOG_LEVEL for debug build

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

**Review by gjc13 — APPROVED** [review]

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: third_party/openthread/CMakeLists.txt [hunk 1]
```diff
@@ -41,13 +41,20 @@ set(OT_PLATFORM_NETIF ON CACHE BOOL "enable platform netif" FORCE)
 set(OT_PLATFORM_UDP ON CACHE BOOL "enable platform UDP" FORCE)
 set(OT_UDP_FORWARD OFF CACHE BOOL "diable udp forward" FORCE)
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(OT_LOG_LEVEL "DEBG" CACHE STRING "set OpenThread log level to DEBG" FORCE)
+else()
+    set(OT_LOG_LEVEL "INFO" CACHE STRING "set OpenThread log level to INFO" FORCE)
+endif()
+
+set(OT_LOG_LEVEL_DYNAMIC ON CACHE BOOL "enable dynamic log level control" FORCE)
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:50]: keep sorted with other options?

**Review by gjc13 — APPROVED** [review]

File: third_party/openthread/CMakeLists.txt (hunk 1 above)
Comment (librasungirl) [reply third_party/openthread/CMakeLists.txt:44]: reordered

File: third_party/openthread/CMakeLists.txt [hunk 2]
```diff
@@ -34,20 +34,28 @@ set(OT_COMMISSIONER ON CACHE BOOL "enable commissioner" FORCE)
...
 set(OT_MAC_FILTER ON CACHE STRING "enable MAC filter" FORCE)
 set(OT_MBEDTLS_CONFIG_FILE "\"${PROJECT_SOURCE_DIR}/third_party/openthread/mbedtls-config.h\"" CACHE STRING "use mbedtls-config.h of this project" FORCE)
 set(OT_PLATFORM "posix" CACHE STRING "use posix platform" FORCE)
 set(OT_PLATFORM_NETIF ON CACHE BOOL "enable platform netif" FORCE)
 set(OT_PLATFORM_UDP ON CACHE BOOL "enable platform UDP" FORCE)
 set(OT_UDP_FORWARD OFF CACHE BOOL "diable udp forward" FORCE)
 
+
```
Comment (librasungirl) [inline third_party/openthread/CMakeLists.txt:52]:
```suggestion
```

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**